netty/src/main/java/io/grpc/netty/NettyServerHandler.java

```diff
@@ -304,6 +309,19 @@ private void cancelStream(ChannelHandlerContext ctx, CancelServerStreamCommand c
     encoder().writeRstStream(ctx, cmd.stream().id(), Http2Error.CANCEL.code(), promise);
   }
 
+  private void verifyContentType(int streamId, Http2Headers headers) throws Http2Exception {
+    ByteString contentType = headers.get(CONTENT_TYPE_HEADER);
+    if (contentType == null) {
+      throw Http2Exception.streamError(streamId, Http2Error.REFUSED_STREAM,
```

Comment:
We should respond with a clean status (via headers). We _must_ not use REFUSED_STREAM though, as that would imply that the client should re-issue the request.

> The REFUSED_STREAM error code can be included in a RST_STREAM frame to indicate that the stream is being closed prior to any processing having occurred. Any request that was sent on the reset stream can be safely retried.